crates/consensus/src/alloy_compat.rs

```diff
@@ -36,6 +38,46 @@ impl TryFrom<AnyTxEnvelope> for OpTxEnvelope {
     }
 }
 
+impl TryFrom<AnyRpcTransaction> for OpTxEnvelope {
+    type Error = ConversionError;
+
+    fn try_from(tx: AnyRpcTransaction) -> Result<Self, Self::Error> {
+        let WithOtherFields { inner: AlloyRpcTransaction { inner, from, .. }, other: _ } = tx;
+
+        match inner {
+            AnyTxEnvelope::Ethereum(TxEnvelope::Legacy(tx)) => {
+                Ok(OpTxEnvelope::Legacy(tx.try_into().map_err(|_| {
+                    ConversionError::Custom("Failed to convert Legacy Tx".to_string())
+                })?))
+            }
+            AnyTxEnvelope::Ethereum(TxEnvelope::Eip2930(tx)) => {
+                Ok(OpTxEnvelope::Eip2930(tx.try_into().map_err(|_| {
+                    ConversionError::Custom("Failed to convert Eip2930 Tx".to_string())
+                })?))
+            }
+            AnyTxEnvelope::Ethereum(TxEnvelope::Eip1559(tx)) => {
+                Ok(OpTxEnvelope::Eip1559(tx.try_into().map_err(|_| {
+                    ConversionError::Custom("Failed to convert Eip1559 Tx".to_string())
+                })?))
+            }
+            AnyTxEnvelope::Ethereum(TxEnvelope::Eip7702(tx)) => {
```

Comment:
I think this can be simplified now that we have a tryinto for TxEnvelope:

https://github.com/alloy-rs/op-alloy/blob/0c6d6dbb2da0a13f81643759046410190e2c59e4/crates/consensus/src/transaction/envelope.rs#L361-L361